dot/state/transaction.go

```diff
@@ -28,8 +30,9 @@ type TransactionState struct {
 
 // NewTransactionState returns a new TransactionState
 func NewTransactionState(telemetry telemetry.Client) *TransactionState {
+	const popTryWait = 10 * time.Millisecond
```

Comment:
Now comes the question... what should this value be? I guess we don't want to wait longer than the slot time... Also even if it's half of the slot time, that means we could miss some extrinsics whilst we sleep. IMO time based is not a solution here.